Clarify two guides and two intro pages

ptgott opened the pull request:
See #11841

Make the Adding Nodes guide more usable for Cloud

- Add a ca_pin preset
- Clarify that tctl must be run on the local machine for Cloud users
- Structure the guide as a step-by-step tutorial. The guide already
  included sequences of sample commands, so all this took was to rename
  headings according to the "Step n/d." format and move the CA pinning
  section into the section on starting the Node.
- Add environment variables to use for storing a CA pin and invite
  token to sample commands, plus piped commands to extract these
  strings from the output of tctl commands.
- Use a ScopedBlock to hide the Node Tunneling section for Cloud users
- Indicate that the --auth-server flag in "teleport start" requires a
  port.

Add intros to the Admin and Operations menu pages

- Clarify the purpose of the Admin Guides and Operations sections
  by adding an intro paragraph to each page. Since these sections are
  similar in scope, I added links from one to the other with statements
  about how the two sections differ.

- Replace lists of links with Tiles.

GitHub SSO

- Move the step to create an OAuth app out of the Prerequisites and
  into its own step. This makes it easier to give the instructions
  to use a specific callback URL proper space.
- Be more explicit about the rp_ip value.
- Add explicit instructions for logging in to the cluster after
  creating the auth preference, including screenshots of expected
  results.

ptgott committed:
Clarify two guides and two intro pages

See #11841

Make the Adding Nodes guide more usable for Cloud

- Add a ca_pin preset
- Clarify that tctl must be run on the local machine for Cloud users
- Structure the guide as a step-by-step tutorial. The guide already
  included sequences of sample commands, so all this took was to rename
  headings according to the "Step n/d." format and move the CA pinning
  section into the section on starting the Node.
- Add environment variables to use for storing a CA pin and invite
  token to sample commands, plus piped commands to extract these
  strings from the output of tctl commands.
- Use a ScopedBlock to hide the Node Tunneling section for Cloud users
- Indicate that the --auth-server flag in "teleport start" requires a
  port.

Add intros to the Admin and Operations menu pages

- Clarify the purpose of the Admin Guides and Operations sections
  by adding an intro paragraph to each page. Since these sections are
  similar in scope, I added links from one to the other with statements
  about how the two sections differ.

- Replace lists of links with Tiles.

GitHub SSO

- Move the step to create an OAuth app out of the Prerequisites and
  into its own step. This makes it easier to give the instructions
  to use a specific callback URL proper space.
- Be more explicit about the rp_ip value.
- Add explicit instructions for logging in to the cluster after
  creating the auth preference, including screenshots of expected
  results.

r0mant commented on docs/pages/setup/admin/adding-nodes.mdx:
```diff
@@ -74,36 +112,53 @@ auth_service:
-### Using Node invitation tokens
+### Start your Node with the invite token and CA pin
+
+The CA pin at the needs to be passed to new Nodes when they're
```
I'm having a hard time parsing "The CA pin at the needs to be passed ...". Is there a typo here?

ptgott replied:
There is—fixed

r0mant commented on docs/pages/setup/admin/adding-nodes.mdx:
```diff
@@ -74,36 +112,53 @@ auth_service:
+starting for the first time, i.e. when they join a cluster.
+
+If a CA pin is not provided, the Teleport Node will join a cluster but it will
+print a `WARN` message (warning).
```
```suggestion
print a warning message.
```

r0mant commented on docs/pages/setup/operations.mdx:
```diff
@@ -4,20 +4,26 @@ description: Teleport Operations - Scaling and High-Availability.
+  Backing up and restoring the cluster.
+  </Tile>
+  <Tile href="./operations/ca-rotation.mdx" title="CA Rotation" icon="wrench">
+  Rotating certificates and certificate authorities.
```
```suggestion
  Rotating Teleport certificate authorities.
```

ptgott committed:
Fix wording and linter issues following PR review

zmb3 commented on docs/pages/setup/admin/adding-nodes.mdx:
```diff
@@ -74,36 +112,53 @@ auth_service:
-### Using Node invitation tokens
+### Start your Node with the invite token and CA pin
+
+A CA pin needs to be passed to new Nodes when they're starting for the first
```
This is a little confusing, because we say it "needs to be passed to new nodes" and then immediately say it doesn't need to be provided to join a cluster.

ptgott committed:
Respond to PR feedback

Remove a misleading instruction re: CA pinning. Also move some text
about CA pinning to a more appropriate location.
